modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -439,6 +448,34 @@ INSFVRhieChowInterpolator::finalize()
 #endif
 }
 
+void
+INSFVRhieChowInterpolator::ghostADataOnBoundary(const BoundaryID boundary_id)
+{
+  if (_a_data_provided || this->n_processors() == 1 ||
+      _velocity_interp_method == Moose::FV::InterpMethod::Average)
+    return;
+
+  // Ghost a for the elements on the boundary
+  for (auto elem_id : _moose_mesh.getBoundaryActiveLocalElemIds(boundary_id))
+  {
+    const auto & elem = _moose_mesh.elemPtr(elem_id);
+    if (elem->processor_id() != this->processor_id())
+      for (const auto i : make_range(_dim))
+        // Adding to the a coefficient will make sure the final result gets communicated
+        addToA(elem, i, 0);
+  }
+
+  // Ghost a for the neighbors of the elements on the boundary
+  for (auto neighbor_id : _moose_mesh.getBoundaryActiveNeighborElemIds(boundary_id))
+  {
+    const auto & neighbor = _moose_mesh.queryElemPtr(neighbor_id);
+    if (neighbor->processor_id() != this->processor_id())
+      for (const auto i : make_range(_dim))
+        // Adding to the a coefficient will make sure the final result gets communicated
+        addToA(neighbor, i, 0);
```

Comment:
per above discussion, let's just do this for component 0

Reply:
done